### ISSUE TITLE
[java] UnusedPrivateMethod false-positive used in method reference

### ISSUE THREAD

**Issue by jordianta** [pmd/pmd#5036]
**Affects PMD Version:** 
7.1.0

**Rule:**
[UnusedPrivateMethod](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedprivatemethod)

**Description:**
Basically, it's the same issue as [this one](https://github.com/pmd/pmd/issues/4817) but when instead of using an explicit lambda we use a method reference

**Code Sample demonstrating the issue:**

This works since the [original issue](https://github.com/pmd/pmd/issues/4817) was fixed:
```java
myRepository.findById(someId).ifPresent(myEntity -> doSomething(myEntity));
```

This doesn't work:
```java
myRepository.findById(someId).ifPresent(this::doSomething);
```

**Expected outcome:**

PMD reports a violation, but that's wrong. That's a false positive. It should work for both scenarios, explicit lambda and method reference.

**Running PMD through:** *Maven* (I tried only with Maven but I guess it happens always)

**Comment by oowekyala** [comment]
Probably fixed by #4989. Ultimately the fact this happens means the analysis classpath is misconfigured. Possibly you have been running pmd before compiling your sources. Does the same thing happen if you do `mvn compile` before `mvn pmd:check`?

See also #4985

**Comment by jordianta** [comment]
Thanks for your quick answer. I've tried to change the order but no luck so far. It's not so straightforward as we use a Springboot bom and I need to change things here and there. Probably I'll wait until the next release (theoretically, end of month) and test with the newer version.

**Comment by oowekyala** [comment]
Will close as duplicate of #4985 for now

**Comment by jordianta-free-now** [comment]
Hi,
After testing with the latest PMD version, 7.2.0, it's still happening:
```
[INFO] 
[INFO] --- pmd:3.22.0:check (validate) @ myservice ---
[INFO] PMD version: 7.2.0
```

```
<violation beginline="124" endline="124" begincolumn="18" endcolumn="41" rule="UnusedPrivateMethod" ruleset="Best Practices" package="com.mycompany.myservice.service" class="MyService" method="myMethod" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.2.0/pmd_rules_java_bestpractices.html#unusedprivatemethod" priority="3">
Avoid unused private methods such as 'myMethod(MyClass)'.
</violation>
```

**Comment by jordianta** [comment]
Hi,
After testing with the latest PMD version, 7.2.0, it's still happening:
```
[INFO] 
[INFO] --- pmd:3.22.0:check (validate) @ myservice ---
[INFO] PMD version: 7.2.0
```

```
<violation beginline="124" endline="124" begincolumn="18" endcolumn="41" rule="UnusedPrivateMethod" 
ruleset="Best Practices" package="com.mycompany.myservice.service" class="MyService" method="myMethod" 
externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.2.0/pmd_rules_java_bestpractices.html#unusedprivatemethod" 
priority="3">
Avoid unused private methods such as 'myMethod(MyClass)'.
</violation>
```
